src/torrent.cpp

```diff
@@ -9227,22 +9227,31 @@ namespace libtorrent
 
 	void torrent::queue_up()
 	{
+		// fix async position change calls for race conditions on changed torrents
+		if(!m_auto_managed || m_abort || is_finished()) return;
+
 		set_queue_position(queue_position() == 0
 			? queue_position() : queue_position() - 1);
 	}
 
 	void torrent::queue_down()
 	{
+		// fix async position change calls for race conditions on changed torrents
+		if(!m_auto_managed || m_abort || is_finished()) return;
+
 		set_queue_position(queue_position() + 1);
 	}
 
 	void torrent::set_queue_position(int p)
 	{
 		TORRENT_ASSERT(is_single_thread());
+
+		// fix async position change calls for race conditions on changed torrents
+		if ((!m_auto_managed || m_abort || is_finished()) && p != -1) return;
```

Comment:
I think you meant to add something like this to ``torrent_handle``. but clearly you can't know the state of the torrent at that point. the only thing you can know is that queue positions < 0 are invalid.